raffle/raffle.py

```diff
@@ -0,0 +1,804 @@
+
+    @raffle.command()
+    async def end(self, ctx: Context, raffle: str):
+        """End a raffle."""
+        msg = await ctx.send(f"Ending the `{raffle}` raffle...")
+        await ctx.trigger_typing()
+        async with self.config.guild(ctx.guild).raffles() as r:
+            raffle_data = r.get(raffle, None)
+            if not raffle_data:
+                return await ctx.send("There is not an ongoing raffle with the name `{}`.".format(raffle))
+            raffle_entities = lambda x: raffle_data[0].get(x)
+            if not ctx.author.id == raffle_entities("owner"):
+                return await ctx.send("You are not the owner of this raffle.")
+            r.pop(raffle)
+        with contextlib.suppress(discord.NotFound):
+            await msg.delete()
+        await ctx.send("Raffle ended.")
+        await self.replenish_cache(ctx)
+    
+    @raffle.command()
+    async def kick(self, ctx: Context, raffle: str, member: discord.Member):
+        """Kick a user from your raffle."""
+        await ctx.trigger_typing()
+        async with self.config.guild(ctx.guild).raffles() as r:
+            raffle_data = r.get(raffle, None)
+            if not raffle_data:
+                return await ctx.send("There is not an ongoing raffle with the name `{}`.".format(raffle))
+            raffle_entities = lambda x: raffle_data[0].get(x)
+            if not ctx.author.id == raffle_entities("owner"):
+                return await ctx.send("You are not the owner of this raffle.")
+            if member.id not in raffle_entities("entries"):
+                return await ctx.send("This user has not entered this raffle.")
+            raffle_entities("entries").remove(member.id)
+            await ctx.send("User removed from the raffle.")
+        await self.replenish_cache(ctx)
+        
+    @raffle.command(name="list")
+    async def _list(self, ctx: Context):
+        """List the currently ongoing raffles."""
+        await ctx.trigger_typing()
+        r = await self.config.guild(ctx.guild).raffles()
+        if not r:
+            return await ctx.send("There are no ongoing raffles.")
+        lines = []
+        for k, v in sorted(r.items()):
+            description = v[0].get("description", None)
+            if not description:
+                description=""
+            lines.append("**{}** {}".format(k, RaffleManager.shorten_description(description)))
+        embeds = []
+        data = list(cf.pagify("\n".join(lines), page_length=1024))
+        for index, page in enumerate(data, 1):
+            embed = discord.Embed(
+                title="Current raffles",
+                description=page,
+                color=await ctx.embed_colour()
+            )
+            embed.set_footer(text="Page {}/{}".format(index, len(data)))
+            embeds.append(embed)
+        await self.compose_menu(ctx, embeds)
+        await self.replenish_cache(ctx)
+
+    @raffle.command()
+    async def teardown(self, ctx: Context):
```

Comment:
This should be restricted. It would not be fun for some troll to just remove every raffle